packages/grid_client/src/clients/tf-grid/twins.ts

```diff
@@ -10,10 +10,22 @@ class Twins {
     this.tfclient = client;
   }
 
-  getPublicKey(mnemonic: string) {
-    const seed = bip39.mnemonicToSeedSync(mnemonic);
-    const privKey = new Uint8Array(seed).slice(0, 32);
-    const pk = "0x" + Buffer.from(secp.getPublicKey(privKey, true)).toString("hex");
+  getPublicKey(secret: string) {
+    let privKey;
+    let pk;
+
+    if (bip39.validateMnemonic(secret)) {
+      const seed = bip39.mnemonicToSeedSync(secret);
+      privKey = new Uint8Array(seed).slice(0, 32);
+      pk = "0x" + Buffer.from(secp.getPublicKey(privKey, true)).toString("hex");
+    } else {
+      if (secret.startsWith("0x")) {
+        secret = secret.substring(2);
+      }
+      privKey = Buffer.from(secret, "hex");
```

Comment:
this is not protected

Reply:
This is already handles by the third party package that encrypts the private key. 

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/40770501/34a9902c-6547-497a-a6f4-2ac60cb1abe0)

Reply:
I added seed validation before generating the public key.